src/net.c

```diff
@@ -1048,6 +1061,88 @@ token_match(const uint8_t *a, size_t alen,
   return alen == blen && (alen == 0 || memcmp(a, b, alen) == 0);
 }
 
+int
+coap_client_delay_first(coap_session_t *session)
+{
+  if (session->type == COAP_SESSION_TYPE_CLIENT && session->doing_first) {
+#ifndef WITH_LWIP
+    if (session->doing_first) {
+      unsigned ref;
+      int timeout_ms = 5000;
+      /*
+       * Need to wait for first request to get out and response back before
+       * continuing.. Response handler has to clear doing_first if not an error.
+       */
+      ref = ++session->ref;
+      while (*(&session->doing_first) != 0) {
+        int result = coap_io_process(session->context, 1000);
+
+        if (result < 0) {
+          session->doing_first = 0;
+          coap_session_release(session);
+          return 0;
+        }
+        if (result <= timeout_ms) {
+          timeout_ms -= result;
+        }
+        else {
+          if (*(&session->doing_first) == 1 && ref > session->ref) {
```

Comment:
Same here: no need for taking the address and then dereference the pointer.

And for the LWIP version below as well.

Reply:
Updated